code/game/objects/effects/mines.dm

```diff
@@ -85,15 +85,39 @@
 	if(foot_on_mine?.resolve())
 		return
 
-	foot_on_mine = WEAKREF(arrived)
+	var/gonna_blow
+	if(istype(arrived, /obj/structure/window))
```

Comment:
Should check for `flags_1 & ON_BORDER_1`? So it covers railings as well (which I presume might have the same issue)